### PR TITLE
[items] Fix error handling for `quantityState`

### PR DESCRIPTION
**Pull request by florian-h05**
HistoricItem creation failed for Items that were not Quantity-compatible because the catch mechanism for QuantityError had a design issue.

Signed-off-by: Florian Hotze <florianh_dev@icloud.com>

I have verified that this fix is working.

### CODE REVIEW

**Comment by stefan-hoehn** [comment]
I checked the fix and it is working correctly for me. The error has gone away.

**Comment by florian-h05** [comment]
@digitaldan Can you please review and merge? The other PR as well, they are both relatively short.

File: items/item-history.js [hunk 1]
```diff
@@ -408,8 +409,12 @@ class ItemHistory {
     try {
       quantityState = Quantity(rawState.toString());
     } catch (e) {
-      if (e instanceof QuantityError) quantityState = null;
-      throw Error('Failed to create "quantityState": ' + e);
+      if (e instanceof QuantityError) {
+        log.debug(`Failed to create "quantityState" for Item ${this.name}:`, e);
```
Comment (digitaldan) [inline items/item-history.js:413]: Do we want to leave debug messages like this here?  I wonder if this is just adding noise?

**Review by digitaldan — APPROVED** [review]
I just had one comment about the debug messages,  I personally would error on being silent, unless its something like a depreciation warning which requires user intervention (and would be removed eventually)

Other then that , LGTM

File: items/item-history.js (hunk 1 above)
Comment (florian-h05) [reply items/item-history.js:413]: I agree, I will remove this logging.

**Pull request merged**